src/measurements.rs

```diff
@@ -1,21 +1,40 @@
 use ndarray::{s, Array3, ArrayBase, Axis, Data, Ix3, Zip};
+use num_traits::{Bounded, FromPrimitive, Num, NumAssignOps, ToPrimitive, Unsigned};
 
 use crate::Mask;
 
-const BACKGROUND: u16 = 0;
-const FOREGROUND: u16 = 1;
+pub trait LabelType:
+    Copy + Num + FromPrimitive + ToPrimitive + Ord + Unsigned + NumAssignOps + Bounded
+{
+    fn background() -> Self;
+    fn foreground() -> Self;
+}
+
+impl<T> LabelType for T
+where
+    T: Copy + Num + FromPrimitive + ToPrimitive + Ord + Unsigned + NumAssignOps + Bounded,
+    std::ops::Range<T>: Iterator,
```

Comment:
I'm not sure why you need `std::ops::Range<T>: Iterator,`...?

Reply:
Me neither, I thought I deleted that 😅